extensions/notebook-renderers/src/textHelper.ts

```diff
@@ -91,22 +93,46 @@ function truncatedArrayOfString(id: string, buffer: string[], linesLimit: number
 
 function scrollableArrayOfString(id: string, buffer: string[], trustHtml: boolean) {
 	const element = document.createElement('div');
-	if (buffer.length > 5000) {
+	if (buffer.length > softScrollableLineLimit) {
 		element.appendChild(generateNestedViewAllElement(id));
 	}
 
-	element.appendChild(handleANSIOutput(buffer.slice(-5000).join('\n'), trustHtml));
+	element.appendChild(handleANSIOutput(buffer.slice(-1 * softScrollableLineLimit).join('\n'), trustHtml));
 
 	return element;
 }
 
-export function createOutputContent(id: string, outputs: string[], linesLimit: number, scrollable: boolean, trustHtml: boolean): HTMLElement {
+export function createOutputContent(id: string, outputText: string, linesLimit: number, scrollable: boolean, trustHtml: boolean): HTMLElement {
 
-	const buffer = outputs.join('\n').split(/\r\n|\r|\n/g);
+	const buffer = outputText.split(/\r\n|\r|\n/g);
 
 	if (scrollable) {
 		return scrollableArrayOfString(id, buffer, trustHtml);
 	} else {
 		return truncatedArrayOfString(id, buffer, linesLimit, trustHtml);
 	}
 }
+
+const outputLengths: Record<string, number> = {};
+
+export function appendScrollableOutput(element: HTMLElement, id: string, appended: string, fullText: string, trustHtml: boolean) {
+	if (!outputLengths[id]) {
+		outputLengths[id] = 0;
+	}
+
+	const buffer = appended.split(/\r\n|\r|\n/g);
+	const appendedLength = buffer.length + outputLengths[id];
+	// Allow the output to grow to the hard limit then replace it with the last softLimit number of lines if it grows too large
+	if (buffer.length + outputLengths[id] > hardScrollableLineLimit) {
+		const fullBuffer = fullText.split(/\r\n|\r|\n/g);
+		outputLengths[id] = Math.min(fullBuffer.length, softScrollableLineLimit);
+		const newElement = scrollableArrayOfString(id, fullBuffer.slice(-1 * softScrollableLineLimit), trustHtml);
+		newElement.setAttribute('output-item-id', id);
+		element.replaceWith(newElement);
+	}
+	else {
+		element.appendChild(handleANSIOutput(buffer.join('\n'), trustHtml));
```

Comment:
is this assuming that the previous output always ended with `\n` or `\r` ? It seems only lines are appended to the DOM, characters don't get appended to the existing last line?

Reply:
good call on checking that, I confirmed the current implementation will append to the same line